### ISSUE TITLE
UI stack for front-end deployment

### ISSUE THREAD

**Issue by chriswilty** [ScottLogic/prompt-injection#388]
Use Typescript-CDK to define a standard S3 bucket / cloudfront / cognito deployment.

CDK example on Project Overwatch:
https://github.com/ScottLogic/project-overwatch/blob/main/cloud/lib/ui-stack.ts

**Notes**
- #385 must be completed first!
- Start with a cloudfront domain, but without Route53.
- Cognito userpool linked to an Azure AD sandbox: might already have been done as part of #385, otherwise see Project Overwatch example, linked above.
- At this stage, manually deploy bundled UI to the bucket, using AWS CLI.
- Remember to empty the bucket and destroy the stack after testing! Do not leave it up until we are certain it is secure.
- Document the process of setting up Azure sandbox in cloud README - see Overwatch repo for guide.

**Questions**
- Auto-authenticate on app load, or a login button? Or start with the latter then progress?

**Out of Scope**
- Pipeline deployment of all stacks will be covered in a subsequent issue.

**Acceptance Criteria**
CDK code for (manually) deploying UI stack.
Stack deployed using CDK CLI, UI deployed to bucket using AWS CLI, deployment and connection with back-end verified, then bucket emptied and stack destroyed using CDK CLI.

**Comment by chriswilty** [comment]
**Progress to date**

I have successfully stood up the UI fronted by Cloudfront, but without any authentication. This part can be merged into a (long-lived) feature branch, and then two other issues spawned:
- Route53 for routing, so that UI and API can be SameSite
- Authentication with Cognito, backed by ScottLogic azure AD
